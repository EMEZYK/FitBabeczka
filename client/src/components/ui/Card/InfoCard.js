import { RecipeInfo } from "./InfoCard.styled";

export const RecipeInfoCard = ({ image, alt }) => {
  return (
    <RecipeInfo>
      <img src={image} alt={alt}></img>
      <p>Å‚atwy</p>
    </RecipeInfo>
  );
};
