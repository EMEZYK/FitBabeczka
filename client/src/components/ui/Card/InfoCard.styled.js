import styled from "styled-components";

export const RecipeInfo = styled.div`
  width: 30%;
  height: 10%;
  border-right: 1px solid #3c3c43;

  img {
    height: 1.5rem;
    width: 1.5rem;
  }

  &:last-child {
    border-right: none;
  }
`;
