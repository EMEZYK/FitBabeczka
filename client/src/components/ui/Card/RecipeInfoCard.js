import { RecipeInfo } from "./RecipeInfoCard.styled";

export const RecipeInfoCard = ({ image, alt, details }) => {
  return (
    <RecipeInfo>
      <img src={image} alt={alt} data-testid="info-card"></img>
      <p>{details}</p>
    </RecipeInfo>
  );
};
