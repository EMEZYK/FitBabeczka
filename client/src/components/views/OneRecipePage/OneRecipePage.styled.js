import styled from "styled-components";
import { Typography } from "../../ui/Typography/Typography.styled";
import { RecipeBox } from "../../ui/RecipeBox/RecipeBox.styled";

export const RecipeContainer = styled.div`
  display: flex;
  align-items: center;
`;

export const ImageContainer = styled(RecipeBox)`
  display: flex;
  flex-wrap: wrap;
  height: 90%;
  width: 30%;
  box-shadow: ${({ theme }) => theme.colors.card.boxShadow};
  justify-content: center;
`;

export const TitleRecipe = styled.h1`
  font-size: ${({ theme }) => theme.fontsize.desktop.xxl}px;
  text-align: left;
`;

export const RecipeShortDescription = styled(Typography)`
  font-size: ${({ theme }) => theme.fontsize.desktop.xs}px;
  text-align: left;
  padding: 1rem 0;
`;
export const RecipePreparation = styled(Typography)`
  font-size: ${({ theme }) => theme.fontsize.desktop.xs}px;
  text-align: left;
  margin-top: 1rem;
`;

export const RecipeHeader = styled(Typography)`
  font-size: ${({ theme }) => theme.fontsize.desktop.m}px;
  text-transform: uppercase;
  font-weight: ${({ theme }) => theme.weight.bold};
  text-align: left;
`;

export const RecipeDescriptionContainer = styled.div`
  display: flex;
  height: auto;
  padding-top: 1rem;
  border-top: 1px solid;
`;

export const DescriptionContainer = styled.div`
  width: 65%;
  height: 100%;
`;

export const IngredientsContainer = styled.div`
  width: 35%;
  height: 100%;
`;

export const IngredientList = styled.ul`
  margin-top: 1rem;
  font-size: ${({ theme }) => theme.fontsize.desktop.xs}px;
`;

export const Ingredient = styled.li`
  list-style: none;
  text-align: left;
`;
