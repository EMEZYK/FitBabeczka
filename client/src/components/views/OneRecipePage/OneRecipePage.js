import { theme } from "../../../theme/theme";
import { FlexWrapper } from "../../global-styles/Flex.styled";
import { RecipeBox } from "../../ui/RecipeBox/RecipeBox.styled";
import { Image } from "../../ui/RecipeBox/RecipeBox.styled";
import {
  RecipePreparation,
  RecipeHeader,
  RecipeContainer,
  Ingredient,
  IngredientList,
  IngredientsContainer,
  DescriptionContainer,
  RecipeShortDescription,
  RecipeDescriptionContainer,
  TitleRecipe,
  ImageContainer,
} from "./OneRecipePage.styled";
import { RecipeInfoCard } from "../../ui/Card/InfoCard";
import { useParams } from "react-router-dom";
import { useState, useEffect } from "react";
import axios from "axios";

export const RecipePage = () => {
  const [recipe, setRecipe] = useState([]);

  const { id } = useParams();

  const fetchGetOneRecipe = async () => {
    await axios({
      url: `/recipes/${id}`,
    })
      .then((response) => {
        setRecipe(response.data);
      })
      .catch((error) => console.error(error));
  };

  useEffect(() => {
    fetchGetOneRecipe();
  }, [id]);

  if (recipe.length === 0) {
    return <div>Loading...</div>;
  }

  console.log("przepis", recipe);

  let recipeLevel = recipe.difficultyLevel;

  switch (recipeLevel) {
    case "Łatwy":
      recipeLevel = "/icons/level.png";
      break;
    case "Średni":
      recipeLevel = "/icons/medium-level.png";
      break;
    case "Trudny":
      recipeLevel = "/icons/high-level.png";
      break;
    default:
      recipeLevel = "/icons/medium-level.png";
  }

  return (
    <RecipeContainer>
      <FlexWrapper height={"80vh"} width={"90%"}>
        <ImageContainer>
          <Image src={recipe.image} />
          <RecipeInfoCard image={"/icons/time.png"} details={recipe.time} />
          <RecipeInfoCard
            image={recipeLevel}
            details={recipe.difficultyLevel}
          />
          <RecipeInfoCard
            image={"/icons/plates.png"}
            details={recipe.servingsNumber}
          />
        </ImageContainer>
        <RecipeBox
          height={"90%"}
          width={"60%"}
          padding={"2rem"}
          color={theme.colors.card.secondary}
          boxShadow={theme.colors.card.boxShadow}
        >
          <TitleRecipe>{recipe.name}</TitleRecipe>
          <RecipeShortDescription>{recipe.description}</RecipeShortDescription>
          <RecipeDescriptionContainer>
            <IngredientsContainer>
              <RecipeHeader>Składniki</RecipeHeader>
              <IngredientList>
                {recipe.ingredients.map((ingredient) => (
                  <Ingredient key={ingredient._id}>{ingredient}</Ingredient>
                ))}
              </IngredientList>
            </IngredientsContainer>
            <DescriptionContainer>
              <RecipeHeader>Przygotowanie</RecipeHeader>
              <RecipePreparation>{recipe.preparation}</RecipePreparation>
            </DescriptionContainer>
          </RecipeDescriptionContainer>
        </RecipeBox>
      </FlexWrapper>
    </RecipeContainer>
  );
};

export default RecipePage;
