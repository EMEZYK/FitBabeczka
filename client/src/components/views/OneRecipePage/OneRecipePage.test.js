import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom";
import { Ingredient } from "./OneRecipePage.styled";
import { RecipeInfoCard } from "../../ui/Card/InfoCard";
import * as renderer from "react-test-renderer";
import "jest-styled-components";

describe("RecipePage components", () => {
  afterEach(cleanup);

  test("Should take a snapshot',", () => {
    const { asFragment } = render(<RecipeInfoCard />);
    expect(asFragment(<RecipeInfoCard />)).toMatchSnapshot();
  });

  const recipe = {
    name: "Ciasto",
    description:
      "Prosty przepis na pyszne babeczki owocowe. Zobacz jakie to proste i przygotuj dla swoich bliskich rewelacyjne babeczki owocowe",
    ingredients: ["2jajka", "3"],
    preparation:
      "Margarynę i jajka wyciągamy wcześniej z lodówki, wszystkie składniki powinny mieć temperaturę pokojową.Białka ubijamy na gęstą, błyszczącą pianę z połową ilości cukru. Piana powinna wyglądać jak na bezę – ma być sztywna.",
    category: "624e0288e496e43e6830718d",
    servingsNumber: 8,
    time: "45min",
    difficultyLevel: "średni",
  };

  test("Check if time icon is displayed", async () => {
    render(
      <RecipeInfoCard
        details={recipe.time}
        image={"/icons/time.png"}
        alt="preparation-time"
      />
    );

    const infoCard = screen.queryByTestId("info-card");

    await waitFor(() => {
      expect(infoCard).toHaveAttribute("src", "/icons/time.png");
    });
  });

  test("Check if difficultyLevel icon is displayed", async () => {
    render(
      <RecipeInfoCard
        details={recipe.time}
        image={"/icons/medium-level.png"}
        alt="difficulty-level"
      />
    );

    const infoCard = screen.queryByTestId("info-card");

    await waitFor(() => {
      expect(infoCard).toHaveAttribute("src", "/icons/medium-level.png");
    });
  });
  test("Check if servingsNumber icon is displayed", async () => {
    render(
      <RecipeInfoCard
        details={recipe.time}
        image={"/icons/plates.png"}
        alt="servings-number"
      />
    );

    const infoCard = screen.queryByTestId("info-card");

    await waitFor(() => {
      expect(infoCard).toHaveAttribute("src", "/icons/plates.png");
    });
  });
  test("Ingredient style has list-style none", () => {
    const ingredient = renderer.create(<Ingredient />).toJSON();
    expect(ingredient).toHaveStyleRule("list-style", "none");
  });
});
