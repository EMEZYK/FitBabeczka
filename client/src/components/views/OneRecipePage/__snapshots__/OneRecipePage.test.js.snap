// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage components Should take a snapshot', 1`] = `
<DocumentFragment>
  .c0 {
  width: 30%;
  border-right: 1px solid #3c3c43;
}

.c0 img {
  height: 1.5rem;
  width: 1.5rem;
}

.c0:last-child {
  border-right: none;
}

@media (max-width:876px) {
  .c0 {
    width: 40%;
  }
}

<div
    class="c0"
  >
    <img
      data-testid="info-card"
    />
    <p />
  </div>
</DocumentFragment>
`;
